<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ModelSelectionPage_Detectron2</class>
 <widget class="QWizardPage" name="ModelSelectionPage_Detectron2">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>450</height>
   </rect>
  </property>
  <property name="title">
   <string>Select Model Architecture</string>
  </property>
  <property name="subTitle">
   <string>Choose the model architecture and backbone for Detectron2.</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <property name="spacing">
    <number>15</number>
   </property>
   <item>
    <widget class="QGroupBox" name="architectureGroupBox">
     <property name="title">
      <string>Architecture</string>
     </property>
     <layout class="QVBoxLayout" name="architectureLayout">
      <item>
       <widget class="QFrame" name="maskRcnnFrame">
        <property name="frameShape">
         <enum>QFrame::Shape::StyledPanel</enum>
        </property>
        <layout class="QVBoxLayout" name="maskRcnnLayout">
         <item>
          <widget class="QRadioButton" name="maskRcnnRadio">
           <property name="font">
            <font>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Mask R-CNN (Recommended)</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="maskRcnnDescription">
           <property name="styleSheet">
            <string notr="true">color: #666; margin-left: 20px;</string>
           </property>
           <property name="text">
            <string>Best balance of speed and accuracy
Good for: General object detection</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="cascadeFrame">
        <property name="frameShape">
         <enum>QFrame::Shape::StyledPanel</enum>
        </property>
        <layout class="QVBoxLayout" name="cascadeLayout">
         <item>
          <widget class="QRadioButton" name="cascadeRadio">
           <property name="text">
            <string>Cascade Mask R-CNN</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="cascadeDescription">
           <property name="styleSheet">
            <string notr="true">color: #666; margin-left: 20px;</string>
           </property>
           <property name="text">
            <string>Higher accuracy, slower inference
Good for: When accuracy is critical</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="backboneGroupBox">
     <property name="title">
      <string>Backbone</string>
     </property>
     <layout class="QVBoxLayout" name="backboneLayout">
      <item>
       <widget class="QComboBox" name="backboneCombo"/>
      </item>
      <item>
       <widget class="QLabel" name="backboneInfo">
        <property name="styleSheet">
         <string notr="true">color: #666;</string>
        </property>
        <property name="text">
         <string>Larger backbones provide better accuracy but are slower</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="estimatesLabel">
     <property name="styleSheet">
      <string notr="true">QLabel { background-color: #f0f0f0; padding: 10px; border-radius: 5px; }</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
